packages/opentelemetry-sdk-trace-base/src/export/BatchSpanProcessorBase.ts

```diff
@@ -63,6 +64,12 @@ export abstract class BatchSpanProcessorBase<T extends BufferConfig> implements
         : env.OTEL_BSP_EXPORT_TIMEOUT;
 
     this._shutdownOnce = new BindOnceFuture(this._shutdown, this);
+
+    if (this._maxExportBatchSize > this._maxQueueSize) {
+      diag.warn('BatchSpanProcessor: maxExportBatchSize must be smaller or equal to maxQueueSize, using default values');
+      this._maxExportBatchSize = DEFAULT_ENVIRONMENT.OTEL_BSP_MAX_EXPORT_BATCH_SIZE;
+      this._maxQueueSize = DEFAULT_ENVIRONMENT.OTEL_BSP_MAX_QUEUE_SIZE;
+    }
```

Comment:
I just went with resetting both variables. But I agree that your solution makes more sense here since it only modifies one variable. Should I go with your suggestion?